### PR TITLE
feat: account hoisting

### PR DESCRIPTION
**Pull request by jxom**
Added the ability to hoist an Account to the Wallet Client.

```diff
import { createWalletClient, http } from 'viem'
import { mainnnet } from 'viem/chains'

const [account] = await window.ethereum.request({ method: 'eth_requestAccounts' })

const client = createWalletClient({
+ account,
  chain: mainnet,
  transport: http()
})
const hash = await client.sendTransaction({
- account,
  to: '0xa5cc3c03994DB5b0d9A5eEdD10CabaB0813678AC',
  value: parseEther('0.001')
})
```

### CODE REVIEW

**Review by roninjin10 — COMMENTED** [review]
nice!

**Review by roninjin10 — APPROVED** [review]

File: site/docs/actions/wallet/sendTransaction.md [hunk 1]
```diff
@@ -34,14 +31,62 @@ const hash = await walletClient.sendTransaction({ // [!code focus:99]
...
+export const [account] = await walletClient.getAddresses()
+// Local Account (Private Key, etc)
+export const account = getAccount(...)
+```
+
+:::
+
+### Account Hoisting
+
+If you do not wish to pass an `account` to every `sendTransaction`, you can also hoist the Account on the Wallet Client (see `config.ts`).
```
Comment (roninjin10) [inline site/docs/actions/wallet/sendTransaction.md:53]: is it worth mentioning you can still override it?

**Review by tmm — APPROVED** [review]

File: src/types/account.ts [hunk 2]
```diff
@@ -27,3 +33,10 @@ export type LocalAccount = {
   ) => Promise<Hash>
   type: 'local'
 }
+
+export type ParseAccount<TAccount extends Account | Address | undefined> =
+  TAccount extends Account
+    ? TAccount
+    : TAccount extends Address
+    ? JsonRpcAccount
+    : undefined
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply src/types/account.ts:42]: That's clever I'm going to start using this it's more readable

[the remark this answers is not in the document]
Comment (jxom) [reply src/types/account.ts:42]: Think we need the `undefined` here.

**Pull request merged**